tests/transactional/verify_undelete_snapshots.pm

```diff
@@ -0,0 +1,67 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify that essential snapsots cannot be deleted, see https://jira.suse.com/browse/SLE-3804
+# Maintainer: QA SLE Functional YaST <qa-sle-yast@suse.de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use transactional qw(rpmver get_utt_packages trup_call);
+
+
+# Verify that essential snapshots cannot be deleted, see https://jira.suse.com/browse/SLE-3804
+sub run {
+    select_console 'root-console';
+
+    my @snapshots_before = (split /\n/, script_output('snapper list --disable-used-space'));
+    my $current_snapshot_before;
+    foreach (@snapshots_before) {
+        if ($_ =~ /(?<current_snapshot_before>^\d+)\*/) { $current_snapshot_before = $+{current_snapshot_before} }
+    }
+
+    # Create some snapshots with a transactional update snapshot in the middle.
+    record_info "Check undel", "Snapshot #1 - Check that essential snapshots cannot be deleted, 
+                                creates snapshots and installs a package";
+    assert_script_run("snapper create -d \"Disposable snapshot #1\"");
+    get_utt_packages;
+    trup_call "ptf install" . rpmver('security');
```

Comment:
The thing is, as explained somewhere, we are using a disk image in this test suite that makes it impossible to simply use the "transactional" command normally, we have to download and install some packages locally. So it sounds simple but then we have to find some packages that will not move or be renamed. Those functions are taking care of exactly this, if we don't use them, we have to re-write similar ones. Another option would be to use another disk image, so not a transactional server, and use directly the transactional update comand. I believe the point of this task is to test things in the context of a transactional server, or at least of a transactional update, the feature request seems to me pretty clear about it.

Reply:
The context of this task is to test with snapper, to use a Transactional Server role or Server role image is our choice of testing. The feature is clear about that IMO. To mix test suites with the two lines above is a bad idea, because those script could fail any time and we are not responsible, if there is not quick solution we could decide to not schedule transactional helper apps test suite, but if you mix things with snapper here then  we will have to cherry-pick what to exclude. Besides that, you just need to create an intermediate snapshot, no need for real package here to install.

Reply:
AC1: snapper correctly reports current and active snapshot in a "transactional update"-system

Reply:
I don't really understand the problem here, I'm using some functions from a library, as we are doing all the time. You probably have an idea in mind that I do not understand. I understand that we could simulate what transactional script is doing, but it looked cleat to me that this had to be tested on a transactional server.

Reply:
> AC1: snapper correctly reports current and active snapshot in a "transactional update"-system

maybe it is semantic, but the part 'updates' refers to the scripts that are used in a transactional server role. We don't need to use those scripts if we can just use snapper, specially in this test suite that is dedicated to snapper. It is just a matter or organization.

Reply:
So I would agree that we could have limited scope of our testing and only use snapper commands, whereas using transactional update is more realistic. Also, we have `transactional_update` module which will show us if we have any issues in the wrapper script, so I would say it's fine to go with this approach as it's already implemented. In case we face  mentioned issues, we can address it and change it to snapper only.

Reply:
I just wanted to be conservative with this because the code in transactional is not the best and in my experience requires now and then some little adjustments but if we are aware about mixing here things I'm fine to go with this solution.